server/controllers/contest.js

```diff
@@ -1,17 +1,19 @@
 const Submission = require("../models/Submission");
 const Contest = require("../models/Contest");
 
-exports.getWinner = async (req, res, next) => {
-
+exports.putWinner = async (req, res, next) => {
   const subId = req.query.winner;
   const { contestId } = req.params;
 
   try {
     await Contest.findByIdAndUpdate(contestId, {
-      winner: subId
+      winner: subId,
+      status: "COMPLETED"
     });
 
-    const contest = await Contest.findById(contestId).populate("winner");
+    const contest = await Contest.findById(contestId)
+      .populate("submissions")
+      .populate("creator", "name email profileUrl");
```

Comment:
This doesn't have same level of detail as the other GET /contest/:id, copy that if you choose to keep setContest(res.data.contest) in your chooseWinner function in ContestDetail